WalletWasabi/Blockchain/Transactions/TransactionStore.cs

```diff
@@ -11,15 +13,16 @@
 
 namespace WalletWasabi.Blockchain.Transactions
 {
-	public class TransactionStore
+	public class TransactionStore : IAsyncDisposable
 	{
+		private bool _disposed;
 		public string WorkFolderPath { get; private set; }
 		public Network Network { get; private set; }
 
 		private Dictionary<uint256, SmartTransaction> Transactions { get; set; }
 		private object TransactionsLock { get; set; }
-		private MutexIoManager TransactionsFileManager { get; set; }
-
+		private IoManager TransactionsFileManager { get; set; }
+		private AsyncLock TransactionsAsyncLock { get; set; } = new AsyncLock();
```

Comment:
When adding a new lock object, it should be mentioned what objects it is supposed to protect so that others can verify it is true.

Is this true:

```suggestion
                /// <remarks>Lock object that synchronizes all access to <see cref="TransactionsFileManager"/>.</remarks>
		private AsyncLock TransactionsAsyncLock { get; set; } = new AsyncLock();
```

?

Reply:
Do you disagree with this? Or do you see a better way?